polyfill.d.ts

```diff
@@ -0,0 +1,30 @@
+/// <reference no-default-lib="true"/>
+
+interface SymbolConstructor {
+  /**
+   * A method that is used with the safe assignment operator `?=`.
+   * Any object that implements the `Symbol.result` method should return `[error, undefined]` or `[null, data]` tuple.
+   */
+  readonly result: unique symbol
+}
+
+interface Function {
+  [Symbol.result]<T extends (...args: any[]) => any>(
```

Comment:
Two additional cases must be covered;
1. function never returns
2. function never resolves

```typescript
interface SymbolConstructor {
  /**
   * A method that is used with the safe assignment operator `?=`.
   * Any object that implements the `Symbol.result` method should return `[error, undefined]` or `[null, data]` tuple.
   */
  readonly result: unique symbol
}

interface Function {
  [Symbol.result]<T extends (...args: any[]) => any>(
    this: T,
    ...args: Parameters<T>
  ): 
    [ReturnType<T>] extends [never] ? readonly [NonNullable<unknown>, never] :
    ReturnType<T> extends Promise<never> ? readonly [NonNullable<unknown>, never] :
    ReturnType<T> extends Promise<infer R>
    ? Promise<
        | readonly [NonNullable<unknown>, undefined]
        | readonly [null, Awaited<R>]
      >
    :
        | readonly [NonNullable<unknown>, undefined]
        | readonly [null, ReturnType<T>]
}

interface Promise<T = unknown> {
  [Symbol.result](): Promise<
    | readonly [NonNullable<unknown>, undefined]
    | readonly [null, Awaited<T>]
  >
}
```

[Tests](https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzgUQB5gKYGMYBoUCOArgIYA2cAvnAGZQQhwDkAAjAJ6YC0WAFuWQwA7AOYYAzgHoiMYGXFMA3AChlAE2xkSUDHBFkIAI3KJlcOMCEwMUGiSy6AyuxCGIZAMIQh4mFCI40Kbm5pIAVGFmIXBhcACCcCAYMLwQanApJPDA4nBE4hjpAO7AKRm8uuIkNLok4uLAIkJJVnAQmFBZQQAGAPwAvN0AdFEhsXFC7G2GAFbY8JnZ4IItMLkput3Oru5DOuJEZDDdicmp6eKph+k6MERQQnDdANo29FD4REIaNJaFALonHrPISHMj4NRZEiAjJEMCCEbRGKSUZwHQkNTeMhTfaHGAALjyQmAxEqLjcZCilFU5ks1ls9l0ADEvjhgN5giFntsKXsJHj-gAeAAqcAwqGs31yAAohnLtCJxISSJNnv8AJRwfoAPjgKvY2ulqPMKRyhOFuGNcDlQwVSrgAAVtCQkvTxCLtaj1YSrc8AErJe5CYWcDAe-5iiXCNS5EEYABuNgjvTRGAxWKmzwAct4s2CSIZBIKvgBrIQQIpCbX4IQJpNwfFWgN3B4hzAeyOSmOO+ggHJh2uJqC6lPozFCbFwbO5-OFsOl8uV6twQf1xtIuDNoNtsPC3Xiru5B29-uCyw1KCbz0blPHhinq3RAA+qfTE8zOaEebIWjnxaEZYVlWnzfBgvy1mo-yPiEL5jhmU6gj++BxEUJClIUgp+tqUEbuY15IuuuHmLBabjpO05frORYLkBy5fD8fyQdBxGvmRmaIeCm6Bq2obhlSNIWFYNh2A4Pb3gUIpakSgFLpy5jcuSuy4kc-zSt6Yl9hJVokW+5Gft+v7UQBi7AUSDEQThSI6WxCFgshqHoWofHRPh1LUsoCBRDQrKyByNBCNKJCEqCrg2Opq6Xp50QpPQRQrhgcXIFA7xqSo5jueYWDeL4U5vNA+DKTAEb9LQQgKTsZB8gcKnSgAjOqURRMANBwNKeWXgAhP0JUcZqUUhFlPjwHw2AloSaCYDggrIMQ5CChwmAQC17X4PpVHzsZtHarqJV+EQGBpVQYryLo-WZdlw0VFgY0oOg8zTbNZDzaGS2ptVeDxUO21SXtB38e5-V1OwQhYLQPnso8-mBcFRChVA6l3ppA51sOcnlLF8WJcl0CpfxUSDTlrzYx8b0ClJ-nlbyhWqfVjW0i1bXE3AXU9WCfWogTl2jeNd1TTNpBPQtGCvStcBrT+BZGTJVbfbt-h-elx0FGjnNwCN1085NMAPQLz2LS1hU1ijssZPLh1uao-XeSDvmQwFQUrrDhhhTDcNo7cQZ6ubAmq0T7wFfyKnk2VPJKYHRV1Q15hNQz7XM91js-uz0Sq+rN0Tfd-NzULIvE6tM4S3+NFLibv3m0rp0cxdatXenvPa1ngsvQb4c1k7Nil2b-2W1EQMg2DNsQ6V0OO3DCMnhJIXO6jZ2pi2jwkN7+PV37+Wk0HJUkA58AU6HlXU5HdMWLHTMs4nZDJwN1dp5rmePXrwvLXnYsF4ZG3S9tO2m-t5cYCdKvX1rrfPm98c4t3em3OGncf7-SAA)